### ISSUE TITLE
Close button not shown after Win+Left/Win+Right

### ISSUE THREAD

**Issue by fxkr** [deadalnix/pixel-saver#16]
The activity bar window close button doesn't get shown for windows maximized to half of the screen with the Win+Left / Win+Right hotkeys.

**Comment by deadalnix** [comment]
Ha yes, I had trouble with that one in the past. The sensible thing to do is really not obvious. What do you think is a good solution to that one ?

**Comment by fxkr** [comment]
I agree that there's no obvious best way.

I think it's best to have one X button for all maximized/half-screen-maximized windows, and making that button control the currently active window. The one problem this has is focus-follows-mouse, but since there is some delay between mouse-over-window and window-gets-activated it should be alright.

(I just wondered how well pixel-saver works with multiple monitors. Can't really test it in the next few days though.)

**Comment by jaroslavrehorka** [comment]
+1 
I'm also having this issue (gnome 3.16)

**Comment by HunterGraubard** [comment]
I believe I'm encountering this issue. Here is what I've noticed:

1. If I have a regular window, and then snap it to the side of the screen using Win+Right: the titlebar disappears, but the minimise/restore/close buttons do not appear in the top panel.

2. If I have a maximised window, and then snap it to the side of the screen using Win+Right: the titlebar remains hidden, and the minimise/restore/close buttons remain in the top panel.

It seems that when you use the snap feature, the titlebar is always removed, but no change occurs to the minimise/restore/close buttons in the top panel and they just remain in whatever state that they were already in.

In my personal opinion, Pixel Saver should absolutely not trigger when snapping a window to the side. A window snapped to the side of the screen is not maximised, and therefore the titlebar should be fully displayed. Snapping is really just a shortcut to resizing and repositioning a window, so it should be treated as any other regular window.

**Comment by fxkr** [comment]
> In my personal opinion, Pixel Saver should absolutely not trigger when snapping a window to the side. It should be in its "off" state. A window snapped to the side of the screen is not maximised, and therefore the titlebar should be fully displayed and the minimise/restore/close buttons should not appear in the top panel. Snapping is really just a shortcut to resizing and repositioning a window, so it should be treated as any other regular window.

I strongly disagree. I have an editor snapped left and a shell snapped right all the time. Together they fill the whole screen. And that's exactly when pixel saver is most useful to me (to see more in editor/shell).

Snapping left/right is a shortcut for making a window fill the entire screen vertically, and pixel saver is about saving vertical space.

**Comment by deadalnix** [comment]
@HunterGraubard You are running into a bug due a behavior change in GTK.
@fxkr the problem is, there is no obvious way to fuse the title bar and the activity bar nicely when window are half maximized.

**Comment by miguejinn** [comment]
It's possible to disable the extension in half-maximized windows?

In maximus extension It's disable by default, in extension.js
`const undecorateHalfMaximised = false;`

**Comment by deadalnix** [comment]
No, sadly. GTK (or is it mutter ?) changed its behavior at some point and it is not controllable.

**Comment by bittner** [comment]
To me the window, even if it's cut in half, is still somewhat "maximized". Therefore it may make sense to keep the Close/Minimize/Maximize buttons in the "global menu" title bar. (Not sure if you, technically, rely on a flag for "maximized" that is set to `false` or so when the window is not fully maximized.)

**Comment by tallero** [comment]
@deadalnix
A possible solution for both half-maximized and on-secondary-monitor windows would be to integrate window controls in the app menu like in [focused window indicator](https://gitlab.com/tallero/focused-window-indicator).

**Comment by sipertruk** [comment]
To keep the decoration in half maximized windows : fb7e1814999ab0af5671328a4569eb28a80374ef
